### DISCUSSION TITLE
Lesson 10: ContractNotFound: No contract deployed

### DISCUSSION THREAD

**Discussion by aloyboi** [smartcontractkit/full-blockchain-solidity-course-py#1368] [Q&A]
I am trying to run get_weth .py & this is the error I am facing.
It is saying the mainnet weth token contract address is not deployed for "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"

![image](https://user-images.githubusercontent.com/78015506/161517581-c6e06f97-d011-482a-98a6-8e922aad6eb5.png)

**brownie-config.yaml**
![image](https://user-images.githubusercontent.com/78015506/161517800-03c4d50e-1c1b-42f1-a09e-83731afded72.png)

**get_weth**
![image](https://user-images.githubusercontent.com/78015506/161518690-a890f11d-fdf5-43ff-b58e-c2c8ca814203.png)

I am using mainnet-fork-dev which was created a few lessons back
![image](https://user-images.githubusercontent.com/78015506/161518071-58313f48-3c42-4e23-a610-95d22c73540a.png)

Another quick question: Why do I have to run Ganache client when I want to run on mainnet-fork-dev? Is it not forked from my Alchemy node? Closing Ganache client  gives me "Unable to launch local RPC client." error.

**Comment by cromewar** [comment]
Hello @aloyboi 
Be sure to have the alchemy configuration to development like this:
![image](https://user-images.githubusercontent.com/20199180/161592408-4ae69d97-8a40-488b-8102-92b9db1433c9.png)

The weth token is this one: 
'0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'
remember to respect the upper and lower case letters.

The reason we use ganache is because we are not deploying to a network with persistency, we are just Mocking the mainnet.

**Reply by aloyboi** [reply]
![image](https://user-images.githubusercontent.com/78015506/161666965-45a861d4-a7f2-4b84-924e-e574c66a4ff5.png)


![image](https://user-images.githubusercontent.com/78015506/161667007-c93e9a65-d976-4d6c-ae0c-b630997e6c8a.png)

still doesnt work after changing

**Reply by cromewar** [reply]
I'll suggest this time delete the mainnet-fork-dev and add it again, look for any typo.

**Reply by aloyboi** [reply]
> I'll suggest this time delete the mainnet-fork-dev and add it again, look for any typo.

Did it but still failed.
![image](https://user-images.githubusercontent.com/78015506/161792706-4a6cd53d-3ddd-4649-9bd7-85ce56b11e27.png)

Additionally I found that it was not a problem for a specific contract, it was a problem for any contract I tried to deploy, same issue

**Reply by cromewar** [reply]
Seems the weth token addresses have changed check this out to the get the new ones: https://docs.aave.com/faq/testing-aave

**Reply by aloyboi** [reply]
does not seem like the weth token address changed, either that or I cannot find the new address. 

The code runs smoothly on Kovan network but not on mainnet fork. 

Does this have to do with funding of the mainnet-fork account? Where does it even get its funds?

**Reply by cromewar** [reply]
No I think is due to the address, as mainnetfork is using the weth token deployed on mainnet I know that token changes between versions of aave and on version v2 is different, cannot find either which one is it.

**Reply by aloyboi** [reply]
how do i tell which version of aave I should be using?

from what I am seeing seems like Ethereum mainnet is not available on V3 yet? So we should be using v2, & the weth token address looks the same

**Reply by cromewar** [reply]
I'll try to make more research on this, @PatrickAlphaC a hand on this one pal?
